must-gather/collection-scripts/gather

```diff
@@ -17,32 +20,40 @@ for ns in $(oc get dataprotectionapplications.oadp.openshift.io --all-namespaces
 for ns in ${namespaces[@]}; do
     echo "[namespace=${ns}] Running oc adm inspect"
     /usr/bin/oc adm inspect --dest-dir must-gather/clusters/${clusterID} --all-namespaces ns/${ns} &
+    pids+=($!)
   done
 
 # Collect DPA CRs
 echo "Gathering DPA CRs for namespaces [${namespaces[@]}]"
 /usr/bin/gather_crs ${clusterID} ${namespaces[@]} &
+pids+=($!)
 
 # Collect the logs"
 echo "[cluster=${clusterID}] Gathering logs for namespaces [${namespaces[@]}]"
 /usr/bin/gather_logs ${clusterID} ${logs_since} ${request_timeout} ${skip_tls} ${namespaces[@]} &
+pids+=($!)
 
 
 # Collect metrics from Prometheus
 if [ -z "$essential_only" ]; then
   echo "[cluster=${clusterID}] Gathering prometheus metrics"
   /usr/bin/gather_metrics &
+  pids+=($!)
 else
   echo "Essential-only must-gather was requested. Skipping prometheus metrics collection"
 fi
 
 # Gather version information
 echo "Gathering version information for top-level debug file"
 /usr/bin/gather_versions "/must-gather/clusters/${clusterID}/oadp-debug-info" ${skip_tls} &
+pids+=($!)
+
+# Check if PID array has any values, if so, wait for them to finish
+if [ ${#pids[@]} -ne 0 ]; then
+    echo "Waiting on subprocesses to finish execution."
```

Comment:
Is it worth since we're tracking PIDs to store what we're waiting on? ie. waiting on X step

Reply:
I think we need to wait for specific pids finish...

Reply:
I meant to say is it also viable to echo the PID and process names so we know what we're stuck on